spec/network_interface/network_interface_rh_spec.rb

```diff
@@ -39,6 +39,9 @@ def result(output, exit_status)
 
   subject(:dhcp_interface) do
     allow(File).to receive(:exist?).and_return(true)
+    stub_path = described_class.allocate.send(:path_to_interface_config_file, device_name)
+    allow(Pathname).to receive(:new).and_return(stub_path)
```

Comment:
Isn't this *really* just allow_any_instance_of :wink:

Reply:
I'm sorry this is just hacky as hell haha 😈

Reply:
In review, I think path_to_interface_config_file should just be a class method.  Then you don't need hackery to go about calling a private method on a new instance, but trying to avoid side effects of initialize.